app/assets/javascripts/services/miq_service.js

```diff
@@ -124,4 +124,23 @@ ManageIQ.angular.app.service('miqService', ['$timeout', '$document', '$q', funct
 
     return $q.reject(e);
   };
+
+  this.getProviderTenants = function(callback) {
+    return function(id) {
```

Comment:
@himdel @tzumainn I just happened to notice these functions in `miqService` that are *very* specific to a particular controller.
`miqService` needs to be as generic as possible I think. 

@tzumainn Would it be possible to take these functions out of `miqService` and create a new service (`tenantService` maybe?) that you could inject in the angular controllers that need it?